### PR TITLE
Auto-escalate HTTPS request commands

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
This PR makes commands the attempt to create an HTTPS request automatically escalate to the leader.

To support this, and for general cleanup, the chain of ownership between a bedrock server and its plugins has been changed to be explicit, such that each plugin is owned by a particular server, instead of instantiating a single global object per plugin. This allows the plugin to easily query the server, because it can rely on its owner existing without changing.

There is a co-requisite auth change coming alongside this PR shortly.

All existing tests pass.

### CODE REVIEW

File: BedrockServer.h [hunk 1]
```diff
@@ -213,13 +216,13 @@ class BedrockServer : public SQLiteServer {
     // detached), and shouldn't need to be reset, because the server exits immediately upon seeing this.
     atomic<bool> shutdownWhileDetached;
 
+    // Arguments passed on the command line. This is modified internally and used as a general attribute store.
```
Comment (coleaeason) [inline BedrockServer.h]: this isn't modified internally anymore right?

File: main.cpp [hunk 2]
```diff
@@ -80,18 +80,16 @@ set<string> loadPlugins(SData& args) {
...
 
-    // Instantiate all of our built-in plugins.
-    map<string, BedrockPlugin*> standardPluginMap = {
-        {"DB",     new BedrockPlugin_DB()},
-        {"JOBS",   new BedrockPlugin_Jobs()},
-        {"CACHE",  new BedrockPlugin_Cache()},
-        {"MYSQL",  new BedrockPlugin_MySQL()}
-    };
+    // Register all of our built-in plugins.
+    BedrockPlugin::g_registeredPluginList.emplace(make_pair("DB",    [](BedrockServer& s){return new BedrockPlugin_DB(s);}));
```
Comment (coleaeason) [inline main.cpp]: can you fix the spacing so these are all just a single space before the lambda?

File: test/tests/SSLTest.cpp [hunk 3]
```diff
@@ -1,73 +1,47 @@
...
-        request["Connection"] = "Close";
-        SHTTPSManager::Transaction* t = https.sendRequest("https://www.notarealplaceforsure.com.fake/", request);
-        _wait(t);
-        ASSERT_FALSE(verifyFullResponse(t->response, t->fullResponse));
+    void test() {
+        for (auto& url : (map<string, string>){
+            // Verify we get some HTTP response from google and paypal. We don't care what it is, just that it's valid
+            // HTTP. We want to notice that our fake URL, fails, though.
+            {"www.google.com", "HTTP/1.1"},
+            //{"svcs.paypal.com", "HTTP/1.1"},
```
Comment (coleaeason) [inline test/tests/SSLTest.cpp]: this is a legacy test which could for sure be removed

File: test/tests/SSLTest.cpp [hunk 4]
```diff
@@ -1,73 +1,47 @@
...
+            {"www.notarealplaceforsure.com.fake", "NO_RESPONSE"},
+        }) {
+            SData request("sendrequest");
+            request["Host"] = url.first;
+            request["Connection"] = "Close";
+            request["passthrough"] = "true";
+            vector<SData> results = tester->executeWaitMultipleData({request}, 1);
+            cout << url.first << "> " << results[0].methodLine << ":" << url.second << endl;
```
Comment (coleaeason) [inline test/tests/SSLTest.cpp]: do we need the cout?

**Comment by tylerkaraszewski** [comment]
@coleaeason updated after travis stopped sucking as much.

File: libstuff/SHTTPSManager.h [hunk 5]
```diff
@@ -15,14 +19,21 @@ class SHTTPSManager : public STCPManager {
...
+            const char* what() {
+                return "Can't create SHTTPSManager::Transaction when not leading";
+            }
+        };
     };
 
     // Constructor/Destructor
-    SHTTPSManager();
-    SHTTPSManager(const string& pem, const string& srvCrt, const string& caCrt);
+    //SHTTPSManager();
```
Comment (coleaeason) [inline libstuff/SHTTPSManager.h]: did you mean to leave this commented out default constructor in here? 

Also requiring a plugin to create an shttpsmanager feels like it some what violates the spirit of libstuff, which from the readme is meant to be used just generically for anything c++ related. For example I've made a few projects that source libstuff as an external git module and uses https managers, now thats not possible. 

Should we leave the default constructor for this use case and allow to create an shttpsmanager that doesn't care about leading?

File: test/clustertest/testplugin/TestPlugin.cpp [hunk 6]
```diff
@@ -154,39 +154,48 @@ bool BedrockPlugin_TestPlugin::processCommand(SQLite& db, BedrockCommand& comman
...
-            // If we're calling `process` on a command with a https request, it had better be finished.
-            SASSERT(request->response);
-
-            // Concatenate all of our responses into the body.
-            command.response.content += to_string(request->response) + "\n";
-
-            // If our response is an error, store that.
-            if (request->response >= 400) {
-                allGoodResponses = false;
+        if (command.request.test("passthrough")) {
```
Comment (coleaeason) [inline test/clustertest/testplugin/TestPlugin.cpp]: i don't get what's going on here, can you explain it? or add a comment?

**Comment by coleaeason** [comment]
i think this looks good, but gonna wait to merge until the associated PR is ready

**Pull request merged**